packages/flutter_tools/lib/src/commands/flutter_command_runner.dart

```diff
@@ -113,6 +113,16 @@ class FlutterCommandRunner extends CommandRunner {
   }
   List<BuildConfiguration> _buildConfigurations;
 
+  String get enginePath {
+    if (!_enginePathSet) {
+      _enginePath = _findEnginePath(_globalResults);
+      _enginePathSet = true;
+    }
+    return _enginePath;
+  }
+  String _enginePath;
+  bool _enginePathSet = false;
```

Comment:
A better pattern for this in dart is as follows:

```
  final String enginePath = _initEnginePath();
  String _initEnginePath() { ... }
```

Then you don't need to declare separate storage or manage the _enginePathSet bool manually.  The initializer for the final field will run lazily the first time the getter for the field is called.